### ISSUE TITLE
Build Error: Payload contains two or more files with the same destination path

### ISSUE THREAD

**Issue by CodingCodger** [sarathkcm/Pluralize.NET#19]
I'm porting a Windows Forms app to MAUI.  The app requires pluralizing and singularizing methods.  I attempted to replace:
using System.Data.Entity.Design.PluralizationServices;
with
using Pluralize.NET;

I got the build error mentioned above.

The only other using statement in this .cs file is for System.Text.RegularExpressions.
In another file I have using statements for  Newtonsoft.Json and Newtonsoft.Json.Linq

**Comment by sarathkcm** [comment]
Thank you for reporting this issue. 

Never heard of this error message before, so I did a google search, and there seem to be a few possible solutions for this error.
Popular ones are - clearing the NuGet cache/packages folder, upgrading visual studio,  upgrading the UWP package (sorry, I'm not sure if MAUI needs this package ), or fixing the build target platform/architecture.

I recommend you search as well and try solutions that you find applicable to your scenario. Please come back and let us know if you are able to resolve it somehow - it might help someone else in the future. 

Meanwhile, if you are able to post the build output logs, I will try to have a look.

**Comment by CodingCodger** [comment]
This all started when I was tracking down a bug in my Humanize.cs module.  the "using System.Data.Entity.Design.PluralizationServices;" line of code had an error.  The error message said the Entity namespace did not exist. Therefore, Pluralization Services were not available. I looked around to see if Microsoft had included this service somewhere else, but if they did, I couldn't find where (there were hints its included in a database package.)  Ultimately, I commented out this line.  I went to NuGet and added the Humanize.Core package, I added the required "using Humanizer" line of code in Humanize.cs, and I rewrote the pluralize and singularize routines there.

 

The error only happens when building for a "Windows Machine". Android and IOS build fine.  The error happens regardless of whether building a debug or release version. I applied the new update for VS 2022.  The version went from 17.1 to 17.2. I discovered 730 new compile errors (GridLayout. -> Grid.) Find and replace quickly fixed those. But the APPX1101 error remained.  I removed Humanizer.Core in NuGet and I also removed all associated code. Error still there. I tried deleting the release and debug folders and rebuilding the app, but still no luck.

 

I found only one copy of FPLmp.deps.json in the target folder. I renamed it and built the app again. The file with the original name reappeared. This suggests the file is written to the folder in two different places during build.

 

I tried deleting the NuGet cache to no effect.

 

What follows is the last few lines of the build log.  I can send the whole thing if you happen to be an insomniac.

 

I am not clear on what to try next.  It seems like uninstalling and reinstalling VS 17.2 might be the next step. I’m not looking forward to that.

 

OUTPUT

------

1>C:\Users\drkar.NUC7I5BNH\.nuget\packages\microsoft.windowsappsdk\1.0.0\build\Microsoft.Build.Msix.Packaging.targets(1370,5): error APPX1101: Payload contains two or more files with the same destination path 'FPLmp.deps.json'. Source files: 

1>C:\Users\drkar.NUC7I5BNH\.nuget\packages\microsoft.windowsappsdk\1.0.0\build\Microsoft.Build.Msix.Packaging.targets(1370,5): error APPX1101: C:\Users\drkar.NUC7I5BNH\source\repos\FPLmp\FPLmp\obj\Release\net6.0-windows10.0.19041\win10-x64\FPLmp.deps.json

1>C:\Users\drkar.NUC7I5BNH\.nuget\packages\microsoft.windowsappsdk\1.0.0\build\Microsoft.Build.Msix.Packaging.targets(1370,5): error APPX1101: C:\Users\drkar.NUC7I5BNH\source\repos\FPLmp\FPLmp\bin\Release\net6.0-windows10.0.19041\win10-x64\FPLmp.deps.json

1>Done building project "FPLmp.csproj" -- FAILED.

========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

========== Deploy: 0 succeeded, 0 failed, 0 skipped ==========

**Comment by CodingCodger** [comment]
Just wanted to let you know I solved the problem.  It turned out to be a NuGet package I had installed and forgotten to remove while searching for a workable pluralizer().   Once I deleted the unwanted package, most things returned to normal.  I have a new problem, probably related to upgrading to 17.2, but I don’t think it has anything to do with the humanizer package.  That seems to be working better than the Microsoft version.  Anyway, thanks for the time you spent researching.  It was probably your idea of clearing the NuGet cache that made me think about NuGet packages and eventually lead to the solution.
